ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,173 @@
+import React, { useContext, useMemo, useRef } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { I18nContext } from '../../../contexts/i18n';
+import { useEqualityCheck } from '../../../hooks/useEqualityCheck';
+import { MetaMetricsContext } from '../../../contexts/metametrics.new';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import { updateViewedNotifications } from '../../../store/actions';
+import { UI_NOTIFICATIONS } from '../../../../../shared/notifications';
+import { ETH_SWAPS_TOKEN_OBJECT } from '../../../../../shared/constants/swaps';
+import { BUILD_QUOTE_ROUTE } from '../../../helpers/constants/routes';
+import { getSortedNotificationsToShow } from '../../../selectors';
+
+function getActionFunctions(metricsEvent) {
+  const actionFunctions = {
+    1: () => {
+      metricsEvent({
+        event: 'Swaps Opened',
+        properties: { source: 'Main View', active_currency: 'ETH' },
+        category: 'swaps',
+      });
+      global.platform.openExtensionInBrowser(
+        BUILD_QUOTE_ROUTE,
+        `fromAddress=${ETH_SWAPS_TOKEN_OBJECT.address}`,
+      );
+    },
+    2: () => {
+      global.platform.openTab({
+        url: 'https://metamask.io/download.html',
+      });
+    },
+    3: () => {
+      global.platform.openTab({
+        url:
+          'https://survey.alchemer.com/s3/6173069/MetaMask-Extension-NPS-January-2021',
+      });
+    },
+  };
+
+  return actionFunctions;
+}
+
+export default function WhatsNewPopup({ onClose }) {
+  const metricsEvent = useContext(MetaMetricsContext);
+  const t = useContext(I18nContext);
+
+  const actionFunctions = getActionFunctions(metricsEvent);
+
+  const notifications = useSelector(getSortedNotificationsToShow);
+
+  const contentRef = useRef();
+
+  const memoizedNotifications = useEqualityCheck(notifications);
+  const idRefMap = useMemo(
+    () =>
+      memoizedNotifications.reduce(
+        (_idRefMap, notification) => ({
+          ..._idRefMap,
+          [notification.id]: React.createRef(),
+        }),
+        {},
+      ),
+    [memoizedNotifications],
+  );
+
+  const maybeRenderNotificationItem = (id, isFirstNotification, property) => {
```

Comment:
There are ways this decision tree could be trimmed down using some more clever logic, but I wanted to preserve the explicit intent of the original render blocks

Reply:
This seems more confusing than before :grimacing: These disparate elements don't really relate to each other, and abstracting them behind this "notification item" idea make it harder to follow what's being rendered and why imo.

Reply:
Can you give an example of what you wanted to see per your feedback?

Reply:
The suggestion in [my first comment about this](https://github.com/MetaMask/metamask-extension/pull/10583#discussion_r602258661) was to have two separate methods - one for rendering the first notification, and one for rendering every other one.

That still seems like a reasonable idea to me. But it was just a suggestion - feel free to ignore it if you'd prefer, and go back to how it was before. I just found it difficult to understand is all.

Reply:
change to use two separate methods for rendering first and other notifications is here: 0d8e626b6